position/position.gno

```diff
@@ -312,7 +388,25 @@ func decreaseLiquidity(params DecreaseLiquidityParams) (uint64, string, string,
 		burnPosition(params.tokenId) // just update flag
 	}
 
-	return params.tokenId, params.liquidity.ToString(), _amount0, _amount1, position.poolKey
+	// UNWRAP
+	if pToken0 == consts.WRAPPED_WUGNOT {
+		if params.unwrapResult {
+			// UNWRAP WUGNOT FROM LIQUIDITY
+			unwrap(amount0.Uint64())
+			// UNWRAP WUGNOT FROM FEE
+			unwrap(fee0.Uint64())
+		}
+	}
+	if pToken1 == consts.WRAPPED_WUGNOT {
+		if params.unwrapResult {
+			// UNWRAP WUGNOT FROM LIQUIDITY
+			unwrap(amount1.Uint64())
+			// UNWRAP WUGNOT FROM FEE
+			unwrap(fee1.Uint64())
+		}
+	}
+
+	return params.tokenId, liquidityToRemove, fee0, fee1, amount0, amount1, position.poolKey
```

Comment:
```suggestion
	if !params.unwrapResult {
		return params.tokenId, liquidityToRemove, fee0, fee1, amount0, amount1, position.poolKey
	}
	
	var (
		unwrapAmount  uint64
		unwrapFee uint64
	)
	switch {
	case pToken0 == consts.WRAPPED_WUGNOT:
		unwrapAmount = amount0.Uint64()
    		unwrapFee = fee0.Uint64()
    	case pToken1 == consts.WRAPPED_WUGNOT:
        	unwrapAmount = amount1.Uint64()
        	unwrapFee = fee1.Uint64()
        }
        unwrap(unwrapAmount)
        unwrap(unwrapFee)
        
        return params.tokenId, liquidityToRemove, fee0, fee1, amount0, amount1, position.poolKey
```

I'm not sure if will work this version, but in my opinion, returning early if `unwrapResult` is `false` could reduce the depth of the code.